core/validatorapi/router.go

```diff
@@ -429,9 +428,19 @@ func proposerDuties(p eth2client.ProposerDutiesProvider) handlerFunc {
 			data = []*eth2v1.ProposerDuty{}
 		}
 
+		executionOptimistic, err := getExecutionOptimisticFromMetadata(eth2Resp.Metadata)
```

Comment:
go-eth2-client doesn't support execution optimistic field yet.
Refer: https://github.com/attestantio/go-eth2-client/blob/3087274ef2ac950392c3e362e0d0572aa1c6dada/http/json.go#L12

So I'd suggest to keep the TODO to add execution optimistic field later when they add support for it.